src/chainparams.h

```diff
@@ -58,6 +58,8 @@ class CChainParams
     };
 
     const Consensus::Params& GetConsensus() const { return consensus; }
+    /** Modifiable consensus parameters added by bip135 */
+    Consensus::Params& GetModifiableConsensus() { return consensus; }
```

Comment:
I agree, this can probably be improved by re-using existing code.
Will look into that - I'm not so familiar with the recent Core code and so used some methods from existing implementation.